From 0267f14974442ba5cd16938abd3e88ae4c0087d4 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:00:08 -0400
Subject: [fs] btrfs: fix wrong search path initialization before searching tree root

Message-id: <1407179109-13714-122-git-send-email-sandeen@redhat.com>
Patchwork-id: 87067
O-Subject: [RHEL7.1 PATCH 121/422] Btrfs: fix wrong search path initialization before searching tree root
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Wang Shilong <wangsl.fnst@cn.fujitsu.com>

commit ffcfaf81795471be3c07d6e3143bff31edca5d5a
Author: Wang Shilong <wangsl.fnst@cn.fujitsu.com>
Date:   Wed Jan 15 00:26:43 2014 +0800

    Btrfs: fix wrong search path initialization before searching tree root

    To search tree root without transaction protection, we should neither search commit
    root nor skip locking here, fix it.

    Signed-off-by: Wang Shilong <wangsl.fnst@cn.fujitsu.com>
    Signed-off-by: Josef Bacik <jbacik@fb.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/send.c b/fs/btrfs/send.c
index 84aed2f..aa60cbe 100644
--- a/fs/btrfs/send.c
+++ b/fs/btrfs/send.c
@@ -2095,7 +2095,7 @@ static int send_subvol_begin(struct send_ctx *sctx)
  char *name = NULL;
  int namelen;
 
- path = alloc_path_for_send();
+ path = btrfs_alloc_path();
  if (!path)
   return -ENOMEM;
 
-- 
1.7.1